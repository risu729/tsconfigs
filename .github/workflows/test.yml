name: Test

on:
  pull_request:
    branches:
      - main
  workflow_call:
  workflow_dispatch:
  schedule:
    - cron: "0 0 * * SUN"

permissions: {}

defaults:
  run:
    shell: bash

jobs:
  test:
    runs-on: ubuntu-24.04
    timeout-minutes: 5
    permissions:
      contents: read # for checkout

    outputs:
      tasks: ${{ steps.list.outputs.tasks }}

    steps:
      - name: Checkout
        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
        with:
          persist-credentials: false

      - name: Install mise
        uses: jdx/mise-action@5ac50f778e26fac95da98d50503682459e86d566 # v3.2.0
        with:
          version: 2025.9.25
        env:
          GITHUB_TOKEN: ${{ github.token }}

      - name: Test
        id: list
        run: mise run test

  actions-timeline:
    needs:
      - test
    # skip if the workflow is called from another workflow
    if: ${{ !cancelled() && contains(github.workflow_ref, '/test.yml') }}

    runs-on: ubuntu-24.04
    timeout-minutes: 5
    permissions:
      actions: read

    steps:
      - name: actions-timeline
        # cspell:ignore kesin
        uses: Kesin11/actions-timeline@a7eaabf426cdae26c3582c3fa674b897170dec8f # v2.2.4
