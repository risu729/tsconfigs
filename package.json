{
	"$schema": "https://json.schemastore.org/package.json",
	"name": "@risu729/tsconfigs",
	"version": "0.0.0-development",
	"license": "MIT",
	"author": "risu729",
	"homepage": "https://github.com/risu729/tsconfigs#readme",
	"repository": {
		"type": "git",
		"url": "git+https://github.com/risu729/tsconfigs.git"
	},
	"bugs": {
		"url": "https://github.com/risu729/tsconfigs/issues"
	},
	"devDependencies": {
		"@commitlint/cli": "19.8.0",
		"@commitlint/config-conventional": "19.8.0",
		"@commitlint/cz-commitlint": "19.8.0",
		"@commitlint/types": "19.8.0",
		"@cspell/cspell-types": "8.18.1",
		"@types/bun": "1.2.8",
		"commitizen": "4.3.1",
		"typescript": "5.8.3"
	},
	"type": "module",
	"files": ["presets"],
	"exports": {
		".": "./presets/base.json",
		"./base": "./presets/base.json",
		"./bundler": "./presets/bundler/base.json",
		"./bundler/base": "./presets/bundler/base.json",
		"./tsc": "./presets/tsc/base.json",
		"./tsc/base": "./presets/tsc/base.json",
		"./package.json": "./package.json"
	},
	"publishConfig": {
		"access": "public",
		"provenance": true
	}
}
