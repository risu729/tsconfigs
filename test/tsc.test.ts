import { describe, expect, it } from "bun:test";
import { resolve } from "node:path";
import { $ } from "bun";
import { exports } from "../package.json" with { type: "json" };

const tsconfigPaths: string[] = [
	...new Set(
		Object.entries(exports)
			.filter(([key]) => !key.endsWith("package.json"))
			.map(([_, value]) => value),
	),
];

describe("tsc --showConfig", async () => {
	for await (const path of tsconfigPaths) {
		it(path, async () => {
			const output = await $`bun run tsc --showConfig --project ${path}`.text();
			const rootDir = resolve(path, "..");
			expect(output.replaceAll(rootDir, ".")).toMatchSnapshot();
		});
	}
});

describe("tsc", async () => {
	for await (const path of tsconfigPaths) {
		it(path, async () => {
			await $`bun run tsc --project ${path}`;
		});
	}
});
