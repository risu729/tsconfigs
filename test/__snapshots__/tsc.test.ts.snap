// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`tsc --showConfig ./presets/base.json 1`] = `
"{
    "compilerOptions": {
        "allowUnreachableCode": false,
        "allowUnusedLabels": false,
        "exactOptionalPropertyTypes": true,
        "noFallthroughCasesInSwitch": true,
        "noImplicitOverride": true,
        "noImplicitReturns": true,
        "noPropertyAccessFromIndexSignature": true,
        "noUncheckedIndexedAccess": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "strict": true,
        "module": "preserve",
        "noUncheckedSideEffectImports": true,
        "resolveJsonModule": true,
        "noEmit": true,
        "erasableSyntaxOnly": true,
        "verbatimModuleSyntax": true,
        "lib": [
            "esnext"
        ],
        "moduleDetection": "force",
        "target": "esnext",
        "incremental": true,
        "skipLibCheck": true,
        "moduleResolution": "bundler",
        "isolatedModules": true,
        "esModuleInterop": true,
        "allowSyntheticDefaultImports": true,
        "resolvePackageJsonExports": true,
        "resolvePackageJsonImports": true,
        "preserveConstEnums": true,
        "useDefineForClassFields": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strictNullChecks": true,
        "strictFunctionTypes": true,
        "strictBindCallApply": true,
        "strictPropertyInitialization": true,
        "strictBuiltinIteratorReturn": true,
        "alwaysStrict": true,
        "useUnknownInCatchVariables": true
    },
    "include": [
        "./**/*",
        "./**/.*",
        "./**/.*/**/*",
        "./**/.*/**/.*"
    ]
}
"
`;

exports[`tsc --showConfig ./presets/bundler/base.json 1`] = `
"{
    "compilerOptions": {
        "allowUnreachableCode": false,
        "allowUnusedLabels": false,
        "exactOptionalPropertyTypes": true,
        "noFallthroughCasesInSwitch": true,
        "noImplicitOverride": true,
        "noImplicitReturns": true,
        "noPropertyAccessFromIndexSignature": true,
        "noUncheckedIndexedAccess": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "strict": true,
        "module": "preserve",
        "noUncheckedSideEffectImports": true,
        "resolveJsonModule": true,
        "noEmit": true,
        "erasableSyntaxOnly": true,
        "verbatimModuleSyntax": true,
        "lib": [
            "esnext"
        ],
        "moduleDetection": "force",
        "target": "esnext",
        "incremental": true,
        "skipLibCheck": true,
        "allowImportingTsExtensions": true,
        "allowJs": true,
        "checkJs": true,
        "moduleResolution": "bundler",
        "isolatedModules": true,
        "esModuleInterop": true,
        "allowSyntheticDefaultImports": true,
        "resolvePackageJsonExports": true,
        "resolvePackageJsonImports": true,
        "preserveConstEnums": true,
        "useDefineForClassFields": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strictNullChecks": true,
        "strictFunctionTypes": true,
        "strictBindCallApply": true,
        "strictPropertyInitialization": true,
        "strictBuiltinIteratorReturn": true,
        "alwaysStrict": true,
        "useUnknownInCatchVariables": true
    },
    "include": [
        "./**/*",
        "./**/.*",
        "./**/.*/**/*",
        "./**/.*/**/.*"
    ]
}
"
`;

exports[`tsc --showConfig ./presets/tsc/base.json 1`] = `
"{
    "compilerOptions": {
        "allowUnreachableCode": false,
        "allowUnusedLabels": false,
        "exactOptionalPropertyTypes": true,
        "noFallthroughCasesInSwitch": true,
        "noImplicitOverride": true,
        "noImplicitReturns": true,
        "noPropertyAccessFromIndexSignature": true,
        "noUncheckedIndexedAccess": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "strict": true,
        "module": "nodenext",
        "noUncheckedSideEffectImports": true,
        "resolveJsonModule": true,
        "noEmit": true,
        "erasableSyntaxOnly": true,
        "verbatimModuleSyntax": true,
        "lib": [
            "esnext"
        ],
        "moduleDetection": "force",
        "target": "esnext",
        "incremental": true,
        "skipLibCheck": true,
        "isolatedDeclarations": true,
        "declaration": true,
        "declarationMap": true,
        "outDir": "./dist",
        "moduleResolution": "nodenext",
        "isolatedModules": true,
        "esModuleInterop": true,
        "allowSyntheticDefaultImports": true,
        "resolvePackageJsonExports": true,
        "resolvePackageJsonImports": true,
        "preserveConstEnums": true,
        "useDefineForClassFields": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strictNullChecks": true,
        "strictFunctionTypes": true,
        "strictBindCallApply": true,
        "strictPropertyInitialization": true,
        "strictBuiltinIteratorReturn": true,
        "alwaysStrict": true,
        "useUnknownInCatchVariables": true
    },
    "include": [
        "./src/**/*",
        "./src/**/.*",
        "./src/**/.*/**/*",
        "./src/**/.*/**/.*"
    ],
    "exclude": [
        "./dist"
    ]
}
"
`;
